### ISSUE TITLE
User interface stuck in Deposit wallet

### ISSUE THREAD

**Issue by jooray** [sparrowwallet/sparrow#208]
Hello,

   when I click on my deposit wallet (probably happens in others), I often get into a weird state in the UTXO tab, where I cannot select utxos - I can click on one, I cannot select two (with shift or cmd) on macOS. Also, the amount shown in "Mix selected" is wrong (the UTXOs don't add up). The Clear button does nothing after clicking and when I go to creating premix, it says at least 100k sats is needed - I have more than 100k selected in UTXOs and Mix selected says a different amount, but it is higher than 100k as well.

   Attached is a video of how that looks in practice. 

https://user-images.githubusercontent.com/1028688/134724173-229982d6-3940-4351-944e-4ce7a0005f87.mp4

**Comment by craigraw** [comment]
What version of macOS does this occur on?

Does anything appear in the log when this happens? See Help menu > Show Log File.

**Comment by jooray** [comment]
macOS 11.6 (20G165)

Right now there are no log entries when clicking there.

Also I should mention that I have multiple open wallets.

**Comment by craigraw** [comment]
Strange issue! 

Have you ever seen this on on another table in the app?

Can you try disabling the UTXOs chart (View menu > uncheck Show UTXOs chart)?

**Comment by jooray** [comment]
I found this in the log now:

```
java.lang.NullPointerException: Cannot invoke "javafx.scene.control.TreeItem.getValue()" because the return value of "javafx.scene.control.TreeTablePosition.getTreeItem()" is null
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.lambda$getSelectedEntries$9(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.Iterator.forEachRemaining(Unknown Source)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.getSelectedEntries(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.updateButtons(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.lambda$initializeView$8(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at javafx.controls/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList.callObservers(Unknown Source)
	at javafx.controls/javafx.scene.control.MultipleSelectionModelBase.shiftSelection(Unknown Source)
	at javafx.controls/javafx.scene.control.MultipleSelectionModelBase.shiftSelection(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeTableView$TreeTableViewArrayListSelectionModel$3.handle(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeTableView$TreeTableViewArrayListSelectionModel$3.handle(Unknown Source)
	at javafx.base/javafx.event.WeakEventHandler.handle(Unknown Source)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler$WeakEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at javafx.base/javafx.event.Event.fireEvent(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeItem.fireEvent(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeItem.updateChildren(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeItem.lambda$new$0(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at javafx.base/javafx.collections.ModifiableObservableListBase.add(Unknown Source)
	at java.base/java.util.AbstractList.add(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.control.RecursiveTreeItem.lambda$addChildrenListener$3(Unknown Source)
	at java.base/java.lang.Iterable.forEach(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.control.RecursiveTreeItem.lambda$addChildrenListener$7(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at javafx.base/javafx.collections.ModifiableObservableListBase.addAll(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.WalletUtxosEntry.updateUtxos(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.walletHistoryChanged(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor64.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber$1.run(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.util.concurrent.DirectExecutor.execute(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber.dispatchEvent(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.EventBus.post(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.WalletForm.lambda$notifyIfChanged$4(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
2021-09-25 09:33:23,947 ERROR [JavaFX Application Thread] c.s.s.MainApp [null:-1] Exception in thread "JavaFX Application Thread"
java.lang.IndexOutOfBoundsException: [ fromIndex: 0, toIndex: 1, size: 0 ]
	at javafx.controls/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList.subList(Unknown Source)
	at javafx.base/javafx.collections.ListChangeListener$Change.getAddedSubList(Unknown Source)
	at javafx.controls/com.sun.javafx.scene.control.SelectedItemsReadOnlyObservableList.getAddedElements(Unknown Source)
	at javafx.controls/com.sun.javafx.scene.control.SelectedItemsReadOnlyObservableList.lambda$new$0(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at javafx.controls/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList._endChange(Unknown Source)
	at javafx.controls/javafx.scene.control.MultipleSelectionModelBase$SelectedIndicesList._endChange(Unknown Source)
	at javafx.controls/javafx.scene.control.ControlUtils.updateSelectedIndices(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeTableView$TreeTableViewArrayListSelectionModel.fireCustomSelectedCellsListChangeEvent(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeTableView.sort(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.control.UtxosTreeTable.updateHistory(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.walletHistoryChanged(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor64.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber$1.run(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.util.concurrent.DirectExecutor.execute(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber.dispatchEvent(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.EventBus.post(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.WalletForm.lambda$notifyIfChanged$4(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
2021-09-25 09:33:23,948 ERROR [JavaFX Application Thread] c.s.s.MainApp [null:-1] Exception in thread "JavaFX Application Thread"
java.lang.NullPointerException: Cannot invoke "javafx.scene.control.TreeItem.getValue()" because the return value of "javafx.scene.control.TreeTablePosition.getTreeItem()" is null
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.lambda$getSelectedEntries$9(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.Iterator.forEachRemaining(Unknown Source)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.getSelectedEntries(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.updateButtons(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.lambda$initializeView$8(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at javafx.base/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at javafx.base/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at javafx.base/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at javafx.controls/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList._endChange(Unknown Source)
	at javafx.controls/javafx.scene.control.MultipleSelectionModelBase$SelectedIndicesList._endChange(Unknown Source)
	at javafx.controls/javafx.scene.control.ControlUtils.updateSelectedIndices(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeTableView$TreeTableViewArrayListSelectionModel.fireCustomSelectedCellsListChangeEvent(Unknown Source)
	at javafx.controls/javafx.scene.control.TreeTableView.sort(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.control.UtxosTreeTable.updateHistory(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.UtxosController.walletHistoryChanged(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor64.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber$1.run(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.util.concurrent.DirectExecutor.execute(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Subscriber.dispatchEvent(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Unknown Source)
	at com.google.common@28.2-jre/com.google.common.eventbus.EventBus.post(Unknown Source)
	at com.sparrowwallet.sparrow/com.sparrowwallet.sparrow.wallet.WalletForm.lambda$notifyIfChanged$4(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
```

Just an idea - maybe because I am mixing, a previously selected item in UTXO list disappeared (for example moved from Premix to Postmix) and somehow you refer to the value?

Removing show UTXO chart does not help.

I have only seen this in UTXO table, but right now in Postmix, so it seems it does not matter which wallet I choose.

**Comment by craigraw** [comment]
Fixed (I hope) in a934ffa.

I think your idea is correct. With this change, if a selected item is removed from the table, the selection is now cleared. Still strange that this has never been reported (or seen by me) before - the situation this occurs in is very common. I'm going to close this as fixed as I can't reproduce it, but please return here if this happens again in a future release.
